trailblazer/store/crud/update.py

```diff
@@ -71,24 +73,22 @@ def update_analysis_jobs_from_slurm_jobs(
         """Update analysis jobs from supplied squeue results."""
         if len(squeue_result.jobs) == 0:
             return
+        self.delete_analysis_jobs(analysis)
+
+        session: Session = get_session()
         for job in squeue_result.jobs:
             job.step = reformat_squeue_result_job_step(
                 data_analysis=analysis.data_analysis, job_step=job.step
             )
-
-        self.delete_analysis_jobs(analysis=analysis)
-        analysis.jobs = [
-            Job(
+            new_job = Job(
                 analysis_id=analysis.id,
                 slurm_id=job.id,
                 name=job.step,
                 status=job.status,
                 started_at=job.started_at,
                 elapsed=job.time_elapsed,
             )
-            for job in squeue_result.jobs
-        ]
-        session: Session = get_session()
+            session.add(new_job)
```

Comment:
Same question here - why are we not replacing the old jobs?

Reply:
We are, all the analysis jobs are deleted above. The old logic deleted all jobs and then (unnecessarily) also overwrote an empty list with `analysis.jobs = ...`.